core/controllers/topics_and_skills_dashboard.py

```diff
@@ -117,6 +123,40 @@ def get(self) -> None:
         for topic_summary_dict in topic_summary_dicts:
             topic_summary_dict['classroom'] = topic_classroom_dict.get(
                 topic_summary_dict['id'], None)
+            topic = topic_fetchers.get_topic_by_id(topic_summary_dict['id'])
+            upcoming_chapters_count = 0
+            overdue_chapters_count = 0
+            total_chapters_counts = []
+            published_chapters_counts = []
+            for story_reference in topic.canonical_story_references:
+                story = story_fetchers.get_story_by_id(
+                    story_reference.story_id)
+                nodes = story.story_contents.nodes
+
+                total_chapters_count = len(nodes)
+                published_chapters_count = len(
```

Comment:
Consider if/else here so you can do a single pass over all items rather than 3 passes (see similar comment in a previous PR).

Reply:
Done